File: 1572_DiagonalSum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        matSize = len(mat)
        totalSum = 0
        
        # left pointer and right pointer
        for leftPtr in range(matSize):
            rightPtr = matSize - 1 - leftPtr
            if leftPtr == rightPtr:
                totalSum += mat[leftPtr][rightPtr]
            else:
                totalSum += mat[leftPtr][leftPtr] + mat[leftPtr][rightPtr]
        
        return totalSum
    

File: 1572_DiagonalSum.cpp
class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int matSize = mat.size();
        int sum = 0;

        // left and right ptr
        for (int left = 0, right = matSize - 1; left < matSize; left++, right--) {
            sum += mat[left][left] + mat[left][right];
        }

        // if matSize is odd, subtract the middle element
        if (matSize % 2 == 1) {
            sum -= mat[matSize / 2][matSize / 2];
        }

        return sum;
    }
};